pkg/webhook/admission/webhook.go

```diff
@@ -84,23 +80,30 @@ type Webhook struct {
 }
 
 func (w *Webhook) setDefaults() {
-	if len(w.Name) == 0 {
-		w.Name = "admissionwebhook.k8s.io"
-	}
 	if len(w.Path) == 0 {
+		if len(w.Rules) == 0 || len(w.Rules[0].Resources) == 0 {
+			// can't do defaulting, skip it.
+			return
+		}
 		if w.Type == types.WebhookTypeMutating {
-			w.Path = "/mutatingwebhook"
+			w.Path = "/mutate-" + w.Rules[0].Resources[0]
```

Comment:
For later on, we should consider adding in `.APIGroups[0]`, but we'll merge as-is for now.